src/Block/SharedBlockBlockService.php

```diff
@@ -154,8 +154,12 @@ private function getBlockBuilder(): FormBuilderInterface
             'translation_domain' => 'SonataPageBundle',
             'edit' => 'list',
         ]);
+        $fieldDescription->setAssociationAdmin($this->sharedBlockAdmin);
 
-        return $this->sharedBlockAdmin->getFormBuilder()->create('blockId', ModelListType::class, [
+        return $this->sharedBlockAdmin->getFormContractor()->getFormBuilder(
+            $this->sharedBlockAdmin->getUniqId(),
+            ['data_class' => $this->sharedBlockAdmin->getClass()]
+        )->create('blockId', ModelListType::class, [
```

Comment:
This kind of fix should be applied to other bundles. There is still a lot of code to write to do such fields.

There were 2 problems:
1. Field description didn't had association admin, and it break on a template from sonataAdmin, fixed line 157.
2. `getFormBuilder` does a lot of things, it instantiate al the form for that admin, and it needs a subject, so it breaks. I picked the first lines where it creates de form builder. There are other options to fix this too. One of them is use the original $form from above, but we are wrapping it on a PageFormMapper (For no reason). 

IMO we might refactor this further, and remove PageFormMapper too, it is not needed.

Reply:
What's the issue ? And in which bundle should we apply the same fix ?

It was never reported before

I would say that PageFormMapper was maybe needed prior to https://github.com/sonata-project/SonataBlockBundle/commit/cd2f8f3ad22f377a6ee542c9c040a30d6795c261. But now the AdminFormMapper implements the BlockBundle FormMapper it's not needed.

Does removing this class (in this PR) would simplify your bugfix ?

Reply:
The problem is:

1. Some of our bundles define blocks (and they do require block bundle) 👍 
2. Some of that blocks are EditableBlocks (an interface defined on block bundle) 👍 
3. That interface is only testable HERE, because those editableBlocks are for SonataPageBundle 👎 
4. When we move for example SonataMediaBundle from 3.x to 4.x we migrated those blocks to newer SonataBlock version, but we didn't test those because of point 3. (and we didn't try to unit test AFAIK, will need to check) 👎 
5. Now that we are migrating this bundle, some errors appear.

Those are other examples of the same problem:
https://github.com/sonata-project/SonataMediaBundle/blob/4.x/src/Block/MediaBlockService.php#L210-L226

I don't know if there are a lot of them (I think not), but this kind of dependencies are hard to test.

Reply:
Take a look at my last commit, not using PageFormMapper, allows to \assert for AdminFormMapper and we have getFormBuilder there.